controllers/job_test.go

```diff
@@ -691,3 +693,52 @@ func TestJobControllerPost_Error(t *testing.T) {
 		assert.Equal(t, rr.Body.String(), errInvalidTransitionRequest.Error())
 	})
 }
+
+func TestJobControllerPost_Timeout(t *testing.T) {
+
+	tests := []struct {
+		current data.JobStatus
+		next    data.JobStatus
+		timeout time.Duration
+	}{
+		{current: data.JobQueued, next: data.JobQueued, timeout: 10 * time.Second},
+		{current: data.JobQueued, next: data.JobQueued, timeout: 0 * time.Second},
```

Comment:
```suggestion
		{current: data.JobQueued, next: data.JobInflight, timeout: 10 * time.Second},
		{current: data.JobQueued, next: data.JobInflight, timeout: 0 * time.Second},
```
Are we allowing timeout for any transition? It should be enough to use timeout only when the next state is Inflight, no?

Reply:
Yes, updated now.